src/mame/video/pgm2.cpp

```diff
@@ -128,36 +136,35 @@ inline void pgm2_state::draw_sprite_line(const rectangle &cliprect, int &mask_of
 			mask_offset += 4;
 		}
 
-
-		mask_offset &= m_sprites_mask_mask;
+		mask_offset &= m_sprites_mask.mask();
 
 		if (zoomybit)
 		{
 			if (!flipx)
 			{
-				if (!reverse) draw_sprite_chunk(cliprect, palette_offset, x, realy, sizex, xdraw, pal, maskdata, zoomx_bits, xrepeats, realxdraw, 1, 1);
-				else draw_sprite_chunk(cliprect, palette_offset, x, realy, sizex, xdraw, pal, maskdata, zoomx_bits, xrepeats, realxdraw, -1, -1);
+				if (!reverse) draw_sprite_chunk(dst, dstpri, cliprect, pen, palette_offset, x, realy, sizex, xdraw, pri, maskdata, zoomx_bits, xrepeats, realxdraw, 1, 1);
+				else draw_sprite_chunk(dst, dstpri, cliprect, pen, palette_offset, x, realy, sizex, xdraw, pri, maskdata, zoomx_bits, xrepeats, realxdraw, -1, -1);
 			}
 			else
 			{
-				if (!reverse) draw_sprite_chunk(cliprect, palette_offset, x, realy, sizex, xdraw, pal, maskdata, zoomx_bits, xrepeats, realxdraw, -1, 1);
-				else draw_sprite_chunk(cliprect, palette_offset, x, realy, sizex, xdraw, pal, maskdata, zoomx_bits, xrepeats, realxdraw, 1, -1);
+				if (!reverse) draw_sprite_chunk(dst, dstpri, cliprect, pen, palette_offset, x, realy, sizex, xdraw, pri, maskdata, zoomx_bits, xrepeats, realxdraw, -1, 1);
+				else draw_sprite_chunk(dst, dstpri, cliprect, pen, palette_offset, x, realy, sizex, xdraw, pri, maskdata, zoomx_bits, xrepeats, realxdraw, 1, -1);
 
 			}
 		}
 		else skip_sprite_chunk(palette_offset, maskdata, reverse);
 	}
 }
 
-void pgm2_state::draw_sprites(screen_device &screen, const rectangle &cliprect, uint32_t* spriteram)
+void pgm2_state::draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect, bitmap_ind8 &primap, const uint32_t* spriteram)
 {
-	m_sprite_bitmap.fill(0x8000, cliprect);
+	//printf("frame\n");
```

Comment:
Leftover from something?  At any rate, you should use `logerror` rather than `printf` for diagnostic output like this.